### PR TITLE
Add de_mac language file

### PR DESCRIPTION
**Pull request by Kile**
Hi there!

As it turns out Apple has a surprise in store for people with a German keyboard Mac; quite a few key combinations are different that other German keyboard layouts. I noticed that when trying to get the ducky to type ~.
image

There *is* an option to switch to the standard German keyboard on Mac but the default is apple's German layout. As this is probably a more common problem I thought it may be worth to add a default file to the selection to fix this.

> **Warning** ~~while I fixed most keys, the backslash key in particular requires a three character combination (`OPT` + `SHIFT` + `7`) and I did not know how this would look in this format.~~ This is now fixed

Other than that, all characters in the normal German keyboard file are covered. Here is the before (the white part is what the ducky wrote):
Screenshot 2023-03-24 at 14 22 50

and this is it with my modifications:
Screenshot 2023-03-24 at 14 24 42

I also tried re-sorting the keys in ascending numerical values as that seems to be the layout you use.

Please suggest an edit or comment how I can also make the backslash work and then I hope this can get merged!

### CODE REVIEW

File: languages/de_mac.json [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+  "@": "40,00,0f",
+  "[": "40,00,22",
+  "]": "40,00,23",
+  "\\": "40,02,24",
+  "|": "40,00,24",
+  "{": "40,00,25",
+  "}": "40,00,26",
+  "COMMAND-CTRL-SHIFT": "40,00,64",
+  "COMMAND-CTRL": "40,00,64",
+  "COMMAND-OPTION-SHIFT'": "40,00,64",
```
Comment (dallaswinger) [inline languages/de_mac.json:152]:
``` 
  "COMMAND-CTRL-SHIFT": "40,00,64",
  "COMMAND-CTRL": "40,00,64",
  "COMMAND-OPTION-SHIFT'": "40,00,64",
  ```
  
As defined here, all three of these combinations press the same keys. These should probably be defined differently.

**Comment by dallaswinger** [comment]
in the case that `the backslash key in particular requires a three character combination` and you know that combination it should be as simple as defining the desired key by combining the hex of the known keys, much like how other key combinations already in the file exist

**Comment by dallaswinger** [comment]
@Kile  I'm okay with merging this if you're able to add your caveat warning to the comment of the file please :) worst case this is a step in the right direction and we can improve and fix as needed

**Comment by Kile** [comment]
You mean my backslash warning?
Is there no way to have it press 3 keys at once?

**Comment by Kile** [comment]
> in the case that `the backslash key in particular requires a three character combination` and you know that combination it should be as simple as defining the desired key by combining the hex of the known keys, much like how other key combinations already in the file exist

Alright will do later

File: languages/de_mac.json (hunk 1 above)
Comment (Kile) [reply languages/de_mac.json:152]: I did not change these from the original German layout file

**Comment by Kile** [comment]
@dallaswinger I fixed the backslash and modified the comment (hopefully to what you wanted).
image

As for the COMMAND combinations you mentioned in your review as I said I did not change these from the original German key layout file. I would not be sure what their correct value is, you would have to ask the original author of the German layout file why they are the same.

**Pull request merged**